### PR TITLE
Translatable mapping drivers

### LINKED ISSUE #77: YAML/XML driver mappings for translatable options

### PR DESCRIPTION
**Pull request by szymach**
Resolves #77
TODO:

- [x] YAML driver.
- [x] Reference docs
- Invalid mapping tests. [separate PR]

### CODE REVIEW

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@fsi-open

File: .scrutinizer.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+checks:
+    php:
+        code_rating: true
+        duplication: true
+
+tools:
+    php_cpd: true
+    php_pdepend:
+    excluded_dirs: [vendor]
```
Comment (chives) [inline .scrutinizer.yml]: tutaj Scrutinizer zgłasza błąd

Comment (szymach) [inline .scrutinizer.yml]: błąd kopypasty :/

File: doctrine-extensions-mapping.xsd [hunk 2]
```diff
@@ -13,4 +13,19 @@
         <xs:attribute name="keyLength" type="xs:integer" use="optional"/>
         <xs:attribute name="keyPattern" type="xs:string" use="optional"/>
     </xs:complexType>
+
+    <xs:element name="translatable" type="fsi:translatable"/>
```
Comment (chives) [inline doctrine-extensions-mapping.xsd]: `name="translatable-field"`

File: doctrine-extensions-mapping.xsd [hunk 3]
```diff
@@ -13,4 +13,19 @@
...
+
+    <xs:complexType name="translatable">
```
Comment (chives) [inline doctrine-extensions-mapping.xsd]: `name="translatable-field"`

File: doctrine-extensions-mapping.xsd [hunk 4]
```diff
@@ -13,4 +13,19 @@
...
+        <xs:attribute name="mappedBy" type="xs:string" use="required"/>
+        <xs:attribute name="field" type="xs:string" use="required"/>
+        <xs:attribute name="targetField" type="xs:string" use="optional"/>
+    </xs:complexType>
+
+    <xs:element name="locale" type="fsi:locale"/>
```
Comment (chives) [inline doctrine-extensions-mapping.xsd]: `name="translatable-locale"`

File: doctrine-extensions-mapping.xsd [hunk 5]
```diff
@@ -13,4 +13,19 @@
...
+
+    <xs:complexType name="locale">
```
Comment (chives) [inline doctrine-extensions-mapping.xsd]: `name="translatable-locale"`

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@rn0

**Pull request merged**